plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1645,8 +1648,8 @@ private void finishOptimize(ConnectorSession session, DeltaLakeTableExecuteHandl
 
             long writeTimestamp = Instant.now().toEpochMilli();
 
-            for (Path scannedPath : scannedPaths) {
-                String relativePath = new Path(tableLocation).toUri().relativize(scannedPath.toUri()).toString();
+            for (String scannedPath : scannedPaths) {
+                String relativePath = new Path(tableLocation).toUri().relativize(new Path(scannedPath).toUri()).toString();
```

Comment:
We could try getting rid of the Path usage here by doing some substring comparison.

Reply:
I tried that, as David mentioned in: https://github.com/trinodb/trino/pull/16469#discussion_r1157882075 but it's not trivial as` getUri()` also encodes special characters. I would like to exclude places where URI is used from this PR.

Reply:
Correct, we'll follow up once https://github.com/trinodb/trino/pull/16596 is merged with `ParsedLocation`. The way Hadoop Path and URI interact is complicated.